src/providers/postgres/qgspostgresdataitemguiprovider.cpp

```diff
@@ -0,0 +1,462 @@
+  }
+
+  if ( QgsPGSchemaItem *schemaItem = qobject_cast< QgsPGSchemaItem * >( item ) )
+  {
+    QAction *actionRefresh = new QAction( tr( "Refresh" ), this );
+    connect( actionRefresh, &QAction::triggered, this, [schemaItem] { schemaItem->refresh(); } );
+    menu->addAction( actionRefresh );
+
+    menu->addSeparator();
+
+    QAction *actionRename = new QAction( tr( "Rename Schema…" ), this );
+    connect( actionRename, &QAction::triggered, this, [schemaItem] { renameSchema( schemaItem ); } );
+    menu->addAction( actionRename );
+
+    QAction *actionDelete = new QAction( tr( "Delete Schema" ), this );
+    connect( actionDelete, &QAction::triggered, this, [schemaItem] { deleteSchema( schemaItem ); } );
+    menu->addAction( actionDelete );
+  }
+
+  if ( QgsPGLayerItem *layerItem = qobject_cast< QgsPGLayerItem * >( item ) )
+  {
+    const QgsPostgresLayerProperty &layerInfo = layerItem->layerInfo();
+    QString typeName = layerInfo.isView ? tr( "View" ) : tr( "Table" );
+
+    QAction *actionRenameLayer = new QAction( tr( "Rename %1…" ).arg( typeName ), this );
+    connect( actionRenameLayer, &QAction::triggered, this, [layerItem] { renameLayer( layerItem ); } );
+    menu->addAction( actionRenameLayer );
+
+    if ( !layerInfo.isView )
+    {
+      QAction *actionTruncateLayer = new QAction( tr( "Truncate %1" ).arg( typeName ), this );
+      connect( actionTruncateLayer, &QAction::triggered, this, [layerItem] { truncateTable( layerItem ); } );
+      menu->addAction( actionTruncateLayer );
+    }
+
+    if ( layerInfo.isMaterializedView )
+    {
+      QAction *actionRefreshMaterializedView = new QAction( tr( "Refresh Materialized View" ), this );
+      connect( actionRefreshMaterializedView, &QAction::triggered, this, [layerItem] { refreshMaterializedView( layerItem ); } );
+      menu->addAction( actionRefreshMaterializedView );
+    }
+  }
+}
+
+
+bool QgsPostgresDataItemGuiProvider::deleteLayer( QgsLayerItem *item, QgsDataItemGuiContext )
+{
+  if ( QgsPGLayerItem *layerItem = qobject_cast< QgsPGLayerItem * >( item ) )
+  {
+    const QgsPostgresLayerProperty &layerInfo = layerItem->layerInfo();
+    QString typeName = layerInfo.isView ? tr( "View" ) : tr( "Table" );
+
+    if ( QMessageBox::question( nullptr, tr( "Delete %1" ).arg( typeName ),
+                                QObject::tr( "Are you sure you want to delete %1.%2?" ).arg( layerInfo.schemaName, layerInfo.tableName ),
+                                QMessageBox::Yes | QMessageBox::No, QMessageBox::No ) != QMessageBox::Yes )
+      return false;
+
+    QString errCause;
+    bool res = QgsPostgresUtils::deleteLayer( layerItem->uri(), errCause );
+    if ( !res )
+    {
+      QMessageBox::warning( nullptr, tr( "Delete %1" ).arg( typeName ), errCause );
+      return false;
+    }
+    else
+    {
+      QMessageBox::information( nullptr, tr( "Delete %1" ).arg( typeName ), tr( "%1 deleted successfully." ).arg( typeName ) );
```

Comment:
Just a general comment: I'm very much looking forward to swapping out all these messageboxes for message bars when you're done with this work!

Reply:
that sounds like a great idea :-)